### ISSUE TITLE
Make basic.feature work for upstream and downstream

### ISSUE THREAD

**Issue by praveenkumar** [minishift/minishift#1004]
We need to make sure our basic features works for upstream and downstream without any additional patching. Currently we have below scenario doesn't work with downstream.

``` gherkin
  Scenario: User can list enabled plugins
    Given Minishift has state "Does Not Exist"
    When executing "minishift addons list" succeeds
    Then stdout should contain
     """
     - anyuid         : enabled    P(0)
     - admin-user     : disabled   P(0)
     """
```

**Comment by agajdosi** [comment]
If I move my perspective from addons to whole Minishift I think there is need to design the way how to handle these differences. For particular CDK only features/scenarios we can have `@cdk` tag, but still there will be particular spheres where both Minishift and CDK will differ - list of addons, expected version of oc, list of example apps available in OpenShift... I did some research today to find possible ways of handling this aspect of keeping tests valid for both upstream and downstream.

**Opt. 1 - config file**
One option is to use some sort of config file and for those steps and their definitions which are different for upstream and downstream use values from config. In this case we would keep this config up to date for upstream so `make integration` would stay executable out of the box. For this purpose we probably should use [cucumber.yaml](https://github.com/cucumber/cucumber/wiki/cucumber.yml) file. (We could also probably save some lines/chars from make file when doing this.)

```cucumber
  Scenario: User can list enabled plugins
    Given Minishift has state "Does Not Exist"
    When executing "minishift addons list" succeeds
    Then stdout should contain config:addons_after_
```
This way we loose some readability, but as there are not many differences between Minishift and CDK it could be fine - most steps will be just as they are now.

**Opt. 2 - excluding scenarios by cucumber tags
This is just idea I had which can be maybe done (info for example [here](https://stackoverflow.com/questions/28408962/how-to-exclude-cucumber-tags)). Idea is to have both variants of same scenario for Minishift and CDK next to each other and exclude the one which is not relevant by tags. So we would have set B of basic scenarios tagged `@basic`, then set M of upstream valid scenarios tagged `@minishift` and set C of downstream valid scenarios tagged `@cdk`. For basic scenario with Minishift we would run: `B \ C`, for downstream we would just run `B \ M`. 

This solution seems ideal to me and I can investigate more tomorrow if nobody is against it.

**Comment by agajdosi** [comment]
So logical operations are possible on Godog tags. This solution is working with our current code:
`make integration GODOG_OPTS="--tags='@test&&~@cdk' -format pretty"`. When I created experimental test.feature file next to basic.feature:

```cucumber
@test
Feature: Just test.feature file searching for the purpose of its existence

  Scenario: test
    Given Minishift has state "Does Not Exist"

  @minishift
  Scenario: test and minishift
    Given Minishift has state "Does Not Exist"

  @cdk
  Scenario: test and cdk
    Given Minishift has state "Does Not Exist"
``` 
only 2 scenarios were executed: `Scenario: test` and `Scenario: test and minishift`.

So we can use this path to handle CDK/Minishift variants for particular scenarios. We can insert append of `&&~@cdk` into make file so you can specify `make integration GODOG_OPTS=-tags proxy` and it will result in tags='proxy&&~@cdk' or we can be more transparent and just instruct developers to add `&&~@cdk` manually.

**Comment by hferentschik** [comment]
Merged via pull request #1019
